### PR TITLE
[Namespaced Skill Install] Namespace installed skills by host/owner/repo

### PR DESCRIPTION
**Pull request by erdemtuna**
## Summary

Namespace dependency skills by `host/owner/repo` from the dependency URL to allow same-name skills from different dependencies. This resolves the real-world collision where PAW and Anthropic both export `skill-creator` — craft previously hard-errored, now they coexist at distinct paths.

### Design Decision

The namespace uses the full `host/owner/repo` path (e.g., `github.com/lossyrob/phased-agent-workflow`) rather than the user-chosen alias, because:
- **Globally unique** — cannot collide even across git hosts (GitHub, GitLab, self-hosted)
- **Deterministic** — does not change if user re-adds with a different alias
- **Transitive-ready** — every dep has a URL, transitives work automatically
- **Familiar** — mirrors Go module paths

### Install Layout

```
<target>/
├── github.com/lossyrob/phased-agent-workflow/
│   ├── paw-implement/
│   └── skill-creator/         ← from PAW
└── github.com/anthropics/skills/
    ├── pdf/
    └── skill-creator/         ← from Anthropic — no conflict
```

### Core Install (internal/cli/install.go)
- collectSkillFiles() prefixes map keys with parsed.PackageIdentity() + "/", producing composite keys like github.com/org/repo/skillName
- verifyIntegrity() uses same composite keys; errors on ParseDepURL failure instead of silently skipping (SoT review fix)

### Resolver (internal/resolve/resolver.go)
- Removed detectCollisions() function and its Phase 5 call — namespacing makes cross-dep collisions structurally impossible

### Remove (internal/cli/remove.go)
- All skills from a removed dep are now treated as orphaned — with namespaced paths, each dep's skills have unique disk paths regardless of name overlap (SoT review caught a critical bug in the original orphan detection logic)
- Added cleanEmptyParents() helper to remove empty host/owner/repo/ ancestor directories after skill removal
- ParseDepURL failure warns user instead of silently falling back to flat path

### Installer (internal/install/installer.go)
- **No signature changes** — composite keys work naturally via filepath.Join creating nested directories
- Updated doc comment

## Testing

| Test | What It Covers |
|------|---------------|
| TestInstallCompositeKeys | Two same-name skills install at different namespace paths |
| TestCollectSkillFiles_SameNameDifferentDeps | Core PAW + Anthropic skill-creator scenario |
| TestCollectSkillFiles_NonGitHubHost | gitlab.example.io namespace paths (SC-005) |
| TestResolveSameNameSkillsAllowed | Resolver accepts same-name skills (replaces collision error test) |
| TestCleanEmptyParents x 3 | Full cleanup, stops at non-empty, stops at root |
| TestRunRemove_SharedSkillRetained | Verifies dep-a same-name skill IS removed under namespacing |
| All existing tests | Updated for namespaced key expectations |

All 15 packages pass (go test ./... -count=1).

## Review

Society of Thought final review with 5 specialists across 4 models:

| Specialist | Model | Findings | Status |
|-----------|-------|----------|--------|
| Correctness | GPT 5.2 | Orphan detection bug — bare names vs namespaced paths | Fixed |
| Security | Gemini 3 Pro | Same orphan bug + integrity silent skip | Fixed |
| Edge Cases | Opus 4.6 | Same + ParseDepURL fallback in remove | Fixed |
| Testing | Sonnet 4.5 | Missing cleanEmptyParents + non-GitHub tests | Added |
| Architecture | Sonnet 4 | Composite key pragmatic, acceptable | Noted |

## Breaking Changes

None. No existing craft users to migrate.

## Follow-up

- #27 — Alias symlinks for shorter skill-to-skill references

## Artifacts

- [Spec.md](.paw/work/namespaced-skill-install/Spec.md) — Feature specification with user stories and acceptance criteria
- [WorkShaping.md](.paw/work/namespaced-skill-install/WorkShaping.md) — Design decisions and rationale
- [SoT Reviews](.paw/work/namespaced-skill-install/reviews/) — 5 specialist review reports

---

Generated with [PAW](https://github.com/lossyrob/phased-agent-workflow)

### CODE REVIEW

File: internal/cli/remove_test.go [hunk 1]
```diff
@@ -173,13 +177,15 @@ resolved:
...
+	if _, err := os.Stat(filepath.Join(targetDir, "github.com", "org", "a", "shared-skill")); err == nil {
+		t.Error("dep-a's shared-skill should have been removed (namespaced paths are independent)")
 	}
 
-	// unique-a should be cleaned up
-	if _, err := os.Stat(filepath.Join(targetDir, "unique-a")); err == nil {
+	// unique-a should be cleaned up since only dep-a provided it
+	if _, err := os.Stat(filepath.Join(targetDir, "github.com", "org", "a", "unique-a")); err == nil {
 		t.Error("unique-a should have been removed")
 	}
```
Comment (erdemtuna) [inline internal/cli/remove_test.go]: **Should** (Testing): `TestRunRemove_SharedSkillRetained` validates that dep-a's skills are removed, but doesn't create dep-b's namespaced skill directory on disk to verify it survives the removal. Since the entire justification for removing the `remainingSkills` deduplication logic is that namespaced paths are independent, the test should verify both sides of that independence.

Add to the test setup (after the existing `MkdirAll` calls):
```go
_ = os.MkdirAll(filepath.Join(targetDir, "github.com", "org", "b", "shared-skill"), 0755)
_ = os.WriteFile(filepath.Join(targetDir, "github.com", "org", "b", "shared-skill", "SKILL.md"), []byte("dep-b skill"), 0644)
```

Add assertion after existing assertions:
```go
// dep-b's identically-named skill should survive dep-a removal
if _, err := os.Stat(filepath.Join(targetDir, "github.com", "org", "b", "shared-skill")); err != nil {
	t.Error("dep-b's shared-skill should survive dep-a removal (independent namespaced paths)")
}
```
